### PR TITLE
fix: Goat Plugin + AWS S3 Service error when env vars absent

### PR DESCRIPTION
**Pull request by jnaulty**
# Relates to:

https://github.com/ai16z/eliza/pull/941 and https://github.com/ai16z/eliza/pull/898

Errors when running agent were:
`TypeError: Cannot read properties of null (reading 'getChain')`

```

 ["✓ Service video initialized successfully"] 

Initializing AwsS3Service
 ⛔ ERRORS
   Failed to initialize service aws_s3: 
   {} 

 ⛔ ERRORS
   Error starting agent for character trump: 
   {} 

Error: Missing required AWS credentials in environment variables
    at _AwsS3Service.initialize (file:///home/jnaulty/github.com/ai16z/eliza/packages/plugin-node/dist/index.js:1890:19)
    at AgentRuntime.initialize (file:///home/jnaulty/github.com/ai16z/eliza/packages/core/dist/index.js:3245:31)
    at async startAgent (file:///home/jnaulty/github.com/ai16z/eliza/agent/src/index.ts:344:9)
    at async startAgents (file:///home/jnaulty/github.com/ai16z/eliza/agent/src/index.ts:368:13)
 ⛔ ERRORS
   Error starting agents: 
   {} 

 ["◎ Chat started. Type 'exit' to quit."] 

You:  ["✓ Server running at http://localhost:3000/"] 
```

# Risks

* none, improves usability when specific env variables are not present

# Background

## What does this PR do?

* fixes issue when environment variables are not set for goat + node plugins

## What kind of change is this?

Bug fixes (non-breaking change which fixes an issue)

## Why are we doing this? Any context or related work?

* main branch does not run unless AWS env variables are set and Alchemy API (for GOAT)
* - NodePlugin is now only created when all required AWS credentials are present (AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY, AWS_REGION, AWS_S3_BUCKET)
- GoatPlugin is now only created when ALCHEMY_API_KEY is present
- Added type annotations for plugins to handle undefined cases

These changes prevent unnecessary plugin initialization when required credentials are missing and improve type safety.

Issues introduced in: https://github.com/ai16z/eliza/pull/941 and https://github.com/ai16z/eliza/pull/898

# Documentation changes needed?

none

# Testing

## Detailed testing steps

None, automated tests are fine.

## Discord username
@crypto__john

### CODE REVIEW

**Comment by odilitime** [comment]
Ugh nodePlugin provided a lot of services, we need to fix it by not requiring those.

**Comment by jnaulty** [comment]
it might be wise to split out the AWS S3 into a separate plugin.

**Comment by odilitime** [comment]
Yes I’ll have to go over the history and see what changed it. It didn’t used to be this way

**Comment by jnaulty** [comment]
I believe this was introduced in PR: https://github.com/ai16z/eliza/pull/941

**Comment by jnaulty** [comment]
ok, @odilitime -- I updated the change to load the node plugin (so all the other services can be accessed)
aws s3 service is still registered, but only works if the env vars are present.

File: agent/src/index.ts [hunk 1]
```diff
@@ -384,9 +384,12 @@ export async function createAgent(
...
 
-    const goatPlugin = await createGoatPlugin((secret) =>
-        getSecret(character, secret)
-    );
+    let goatPlugin: any | undefined;
+    if (getSecret(character, "ALCHEMY_API_KEY")) {
+        goatPlugin = await createGoatPlugin((secret) =>
+            getSecret(character, secret)
+        );
+    }
```
Comment (AgustinRamiroDiaz) [inline agent/src/index.ts:392]: nit: In order to not duplicate the secret check, you could do 

```suggestion
    const getGoatPlugin = async () => await createGoatPlugin((secret) =>
        getSecret(character, secret)
    );
```

and then later on line 440 do

```
            getSecret(character, "ALCHEMY_API_KEY") ? getGoatPlugin() : null,
```

**Review by AgustinRamiroDiaz — APPROVED** [review]
This is currently broken on main, it'd be nice to merge this PR to have it fixed :D

**Review by odilitime — APPROVED** [review]

**Pull request merged**